efficient-eagles/early_internet/main/models.py

```diff
@@ -0,0 +1,143 @@
+from django.db import models
+from django.utils.text import slugify
+from django.contrib.auth.models import AbstractUser
+from django.urls import reverse
+
+
+class TimeStampedModel(models.Model):
+    created = models.DateTimeField(auto_now_add=True)
+    modified = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        abstract = True
+
+
+class CustomUser(AbstractUser):
+    profile_img = models.ImageField(upload_to="profile_imgs/", null=True, blank=True)
+    bio = models.TextField(
+        max_length=100, default="Hi, I am a HoneyFeed User", null=True, blank=True
+    )
+
+    def __str__(self):
+        return self.username
+
+
+class Topic(TimeStampedModel):
+    name = models.CharField(max_length=20, default="", blank=False, unique=True)
+
+    def __str__(self):
+        return self.name
+
+
+class Post(TimeStampedModel):
```

Comment:
nice use of delegation. Inheritance and Django work well together. 
good use of magic methods and property